sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -870,11 +868,6 @@ private static void verifyTypesForMatch(Iterable<? extends Symbol> columns, Data
 
     private static class Comparison {
 
-        private static final Set<ComparisonExpression.Type> NEGATING_TYPES = ImmutableSet.of(
-            ComparisonExpression.Type.REGEX_NO_MATCH,
-            ComparisonExpression.Type.REGEX_NO_MATCH_CI,
```

Comment:
They're still negated, I added a default case to the switch statement (https://github.com/crate/crate/pull/6186/files/7bf0e8ecb984c73a43bf8d5c28be02063cd7a1ce#diff-2ae510d647430a141b29fceca71dc92aR949) 

Can also do that in a separate commit if you'd like as it's unrelated to this change.